test/core/tsi/spiffe_ssl_transport_security_test.cc

```diff
@@ -0,0 +1,438 @@
+constexpr absl::string_view kServerCertPath =
+    "test/core/tsi/test_creds/spiffe_end2end/server_spiffe.pem";
+constexpr absl::string_view kServerChainKeyPath =
+    "test/core/tsi/test_creds/spiffe_end2end/leaf_signed_by_intermediate.key";
+constexpr absl::string_view kServerChainCertPath =
+    "test/core/tsi/test_creds/spiffe_end2end/leaf_and_intermediate_chain.pem";
+constexpr absl::string_view kClientSpiffeBundleMapPath =
+    "test/core/tsi/test_creds/spiffe_end2end/client_spiffebundle.json";
+constexpr absl::string_view kServerSpiffeBundleMapPath =
+    "test/core/tsi/test_creds/spiffe_end2end/server_spiffebundle.json";
+
+constexpr absl::string_view kNonSpiffeKeyPath =
+    "test/core/tsi/test_creds/crl_data/valid.key";
+constexpr absl::string_view kNonSpiffeCertPath =
+    "test/core/tsi/test_creds/crl_data/valid.pem";
+constexpr absl::string_view kNonSpiffeCAPath =
+    "test/core/tsi/test_creds/crl_data/ca.pem";
+constexpr absl::string_view kMultiSanKeyPath =
+    "test/core/tsi/test_creds/spiffe_end2end/multi_san.key";
+constexpr absl::string_view kMultiSanCertPath =
+    "test/core/tsi/test_creds/spiffe_end2end/multi_san_spiffe.pem";
+constexpr absl::string_view kInvalidUtf8SanKeyPath =
+    "test/core/tsi/test_creds/spiffe_end2end/invalid_utf8_san.key";
+constexpr absl::string_view kInvalidUtf8SanCertPath =
+    "test/core/tsi/test_creds/spiffe_end2end/invalid_utf8_san_spiffe.pem";
+
+class SpiffeSslTransportSecurityTest
+    : public testing::TestWithParam<tsi_tls_version> {
+ protected:
+  // A tsi_test_fixture implementation.
+  class SslTsiTestFixture {
+   public:
+    SslTsiTestFixture(absl::string_view server_key_path,
+                      absl::string_view server_cert_path,
+                      absl::string_view client_key_path,
+                      absl::string_view client_cert_path,
+                      absl::string_view server_spiffe_bundle_map_path,
+                      absl::string_view client_spiffe_bundle_map_path,
+                      absl::string_view ca_path, bool expect_server_success,
+                      bool expect_client_success_1_2,
+                      bool expect_client_success_1_3) {
+      tsi_test_fixture_init(&base_);
+      base_.test_unused_bytes = true;
+      base_.vtable = &kVtable;
+      server_key_ = grpc_core::testing::GetFileContents(server_key_path.data());
+      server_cert_ =
+          grpc_core::testing::GetFileContents(server_cert_path.data());
+      client_key_ = grpc_core::testing::GetFileContents(client_key_path.data());
+      client_cert_ =
+          grpc_core::testing::GetFileContents(client_cert_path.data());
+      // We set this and it shouldn't matter if we set spiffe bundles
+      root_cert_ = grpc_core::testing::GetFileContents(ca_path.data());
+      if (!server_spiffe_bundle_map_path.empty()) {
+        auto server_map =
+            grpc_core::SpiffeBundleMap::FromFile(server_spiffe_bundle_map_path);
+        EXPECT_TRUE(server_map.ok());
```

Comment:
nit: This should be `ASSERT_TRUE` or `CHECK`, as we will crash on the next line if `server_map.ok()` is false.

Reply:
Thanks!